src/awesome/awesome.rs

```diff
@@ -105,8 +105,34 @@ fn property_setup<'lua>(lua: &'lua Lua, awesome_table: &Table<'lua>) -> rlua::Re
 
 /// Registers a new X property
 /// This actually does nothing, since this is Wayland.
-fn register_xproperty<'lua>(_: &'lua Lua, _: Value<'lua>) -> rlua::Result<()> {
-    warn!("register_xproperty not supported");
+fn register_xproperty<'lua>(lua: &'lua Lua, (name, v_type): (String, String)) -> rlua::Result<()> {
```

Comment:
I think I would implement this purely in Lua (somehow). In awesome, awesome can be restarted while the X11 server continues running, so there it makes sense to save things in the X11 server. For way-cooler, a restart surely loses all the state (right?), so emulating xproperties in Lua is okay.
What do you think?
(No, I don't know how to actually do this; it's just a thought.)

Reply:
We may still need to implement this actually, because we will be relying on xwayland a lot and, AFAIK, the xwayland server lets you register xproperties to the clients still.

If the main use case is to store state in case of a restart though, that means we may need to do some hoops to achieve this...

You're right in that if way-cooler was to restart (e.g the process died) then we would lose all state because Way Cooler is not only the window manager but also the display server. If the use case is primarily recovery from the process dying (e.g from an error) then we would need to achieve this via some other way.

One thing we _could_ do is run Lua in a totally separate process and communicate between Way Cooler and Lua via IPC. Though that would obviously be:

1) Really slow
2) Way more complicated

That would achieve the same thing Awesome currently does with X11 (since they are separate processes there) but it would mean more work.

Depending on the use case, this could drastically affect the design. But it shouldn't affect implementing the interface, as I'll be moving things around when we move to wlroots anyways. I'll be thinking about this though, thank you.

Reply:
Well, the "global state" (xproperties on `root`) is not all that interesting, I think. And xproperties on individual client objects will still be lost since when the display server dies, all clients die with it.

Yes, this xproperty API is from a time when awesome was still restarting whenever the screen configuration changed. Ever since awesome stopped doing that, I haven't seen much use of this API.